src/AppBundle/v2/Registration/Uploader/LayDeputyshipUploader.php

```diff
@@ -0,0 +1,190 @@
+        $this->em = $em;
+        $this->clientRepository = $clientRepository;
+        $this->reportService = $reportService;
+        $this->casRecFactory = $casRecFactory;
+    }
+
+    /**
+     * @param LayDeputyshipDtoCollection $collection
+     * @return array
+     */
+    public function upload(LayDeputyshipDtoCollection $collection): array
+    {
+        $this->throwExceptionIfDataTooLarge($collection);
+
+        try {
+            $this->em->beginTransaction();
+
+            foreach ($collection as $index => $layDeputyshipDto) {
+                if ($this->clientBelongsToDifferentDeputy($layDeputyshipDto)) {
+                    $this->ignored[] = sprintf('%s:%s', $layDeputyshipDto->getCaseNumber(), $layDeputyshipDto->getDeputyNumber());
+                    continue;
+                }
+
+                try {
+                    $this->createAndPersistNewCasRecEntity($layDeputyshipDto);
+                } catch (CasRecCreationException $e) {
+                    $this->errors[] = sprintf('ERROR IN LINE %d: %s', $index + 2, $e->getMessage());
+                    continue;
+                }
+
+                $this->handleBatchDatabaseFlush();
+            }
+
+            $this
+                ->updateReportTypes()
+                ->commitTransactionToDatabase();
+
+        } catch (\Throwable $e) {
+            return ['added' => $this->added, 'errors' => [$e->getMessage()]];
+        }
+
+        return [
+            'added' => $this->added,
+            'errors' => $this->errors,
+            'ignored-count' => count($this->ignored),
+            'ignored' => $this->ignored
+        ];
+    }
+
+    /**
+     * @param LayDeputyshipDtoCollection $collection
+     */
+    private function throwExceptionIfDataTooLarge(LayDeputyshipDtoCollection $collection): void
+    {
+        if ($collection->count() > self::MAX_UPLOAD) {
+            throw new \RuntimeException(sprintf(
+                'Max %d records allowed in a single bulk insert',
+                self::MAX_UPLOAD
+            ));
+        }
+    }
+
+    /**
+     * @param LayDeputyshipDto $layDeputyshipDto
+     * @return bool
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    private function clientBelongsToDifferentDeputy(LayDeputyshipDto $layDeputyshipDto): bool
+    {
+        $result = $this
+            ->clientRepository
+            ->clientIsAttachedButNotToThisDeputy($layDeputyshipDto->getCaseNumber(), $layDeputyshipDto->getDeputyNumber());
+
+        if (false === $result) { return false; }
+
+        // Some deputies have a ',' separated list of deputy nums so if the above query tells us the client is registered to
+        // a different deputy, it may be a false result if the deputy num is concealed with a string list.
+        // Double check for the deputy num within a string list.
+        $deputyNumbers = explode(',', $result['deputy_no']);
+
+        return (in_array($layDeputyshipDto->getDeputyNumber(), $deputyNumbers)) ? false : true;
```

Comment:
Would this be simpler as `!in_array(...)` rather than ternary?

Reply:
Sure, I'll update.